packages/react/src/components/LinkWithIcon/LinkWithIcon.js

```diff
@@ -17,33 +23,120 @@ const { prefix } = settings;
 /**
  * LinkWithIcon component.
  */
-const LinkWithIcon = ({ children, href, ...props }) => {
+const LinkWithIcon = ({
+  children,
+  href,
+  copy,
+  type,
+  direction,
+  visited,
+  ...props
+}) => {
+  const RenderIcon = _renderLinkIcon(type);
+
+  console.log('visited state', visited);
   return (
     <div
-      className={`${prefix}--link-with-icon__container`}
+      className={cx(`${prefix}--link-with-icon__container`, {
+        [`${prefix}--link-with-icon__container--disabled`]: props.disabled,
+      })}
       data-autoid={`${stablePrefix}--link-with-icon`}>
-      <Link href={href} className={`${prefix}--link-with-icon`} {...props}>
+      <Link
+        href={href}
+        className={cx(`${prefix}--link-with-icon`, {
+          [`${prefix}--link-with-icon--visited`]: visited,
+        })}
+        {...props}>
+        {direction === 'left' ? (
+          <RenderIcon className={`${prefix}--link-with-icon_left-icon`} />
+        ) : (
+          ''
+        )}
+        {copy}
         {children}
+        {direction === 'right' ? (
+          <RenderIcon className={`${prefix}--link-with-icon_right-icon`} />
+        ) : (
+          ''
+        )}
       </Link>
     </div>
   );
 };
 
+const _renderLinkIcon = type => {
```

Comment:
Don't think this logic belongs here. I think we already have this kind of logic in CTALogic. Let's keep `LinkWithIcon` as generic as possible.